### PR TITLE
[serve] Remove serve.route decorator

### PR DESCRIPTION
**Pull request by edoakes**
## Why are these changes needed?

This doesn't work properly when imported and we want to keep the API minimal for now.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

**Comment by edoakes** [comment]
I verified that all of the affected examples still run.

File: python/ray/serve/tests/test_persistence.py [hunk 1]
```diff
@@ -14,9 +14,12 @@ def test_new_driver(serve_instance):
 from ray import serve
 serve.init()
 
-@serve.route("/driver")
 def driver(flask_request):
     return "OK!"
+
+serve.create_endpoint("driver", "/driver")
+serve.create_backend(driver, "driver")
+serve.split("driver", {{"driver": 1.0}})
```
Comment (simon-mo) [inline python/ray/serve/tests/test_persistence.py:22]: accidental double {{

**Review by simon-mo — APPROVED** [review]

File: python/ray/serve/tests/test_persistence.py (hunk 1 above)
Comment (edoakes) [reply python/ray/serve/tests/test_persistence.py:22]: This is actually to escape the {{ because it's in a format string

Comment (simon-mo) [reply python/ray/serve/tests/test_persistence.py:22]: i see can you add a comment?

Comment (simon-mo) [reply python/ray/serve/tests/test_persistence.py:22]: oh it's merged. just make sure piggyback it on any follow up PR

**Pull request merged**